### PR TITLE
Flip button updates

### PR DESCRIPTION
**Pull request by jinchung**
### Changes include:
* In order to simplify the trade details calculations, updated the redux state to mainly hold the user-updated value (the rest of the inputs are then derived)
* Hitting max when ETH is the input currency will now always reserve 0.01 ETH (this helps prevent users from not being able to take further actions after the swap, and helps prevent the input number from jumping around when gas prices are updated)
* Changing the input or output currency now behaves as expected (more notes below)
* Clean up of a lot of hooks

### Currency Update Expected Behavior:
**Changing input currency**
Focus: input > Clear
Focus: max > Clear
Focus: native > Use native value to update the rest
Focus: output > Use output value to update the rest

**Changing output currency**
Focus: output > Clear
Focus: input > Use input value to update the rest
Focus: native > Use native value to update the rest
Focus: max > Use input value to update the rest

### CODE REVIEW

File: src/hooks/useSwapIsSufficientLiquidity.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export default function useSwapIsSufficientLiquidity(
+  tradeDetails: Trade | null
+) {
+  const independentValue = useSelector(
+    (state: AppState) => state.swap.independentValue
+  );
+  const inputCurrencyAddress = useSelector(
+    (state: AppState) => state.swap.inputCurrency?.address
+  );
+  const outputCurrencyAddress = useSelector(
```
Comment (mikedemarais) [inline src/hooks/useSwapIsSufficientLiquidity.ts:16]: couldnt you do these all as one selector?

**Review by mikedemarais — APPROVED** [review]
lgtm!

File: src/hooks/useSwapIsSufficientLiquidity.ts (hunk 1 above)
Comment (mikedemarais) [reply src/hooks/useSwapIsSufficientLiquidity.ts:16]: theres another place in this PR that this same comment could apply to

**Comment by jinchung** [comment]
Will address Mike's comments in a separate PR!

**Pull request merged**